### DISCUSSION TITLE
Pytorch Learning Materials

### DISCUSSION THREAD

**Discussion by catunderrain** [community/community#132221] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi everyone,
Due to the needs of my internship, I need to learn (and possibly master) PyTorch. I already have a background in ML and DL algorithms. Could you introduce some materials, documents (books, videos, courses, etc.) that provide both a broad overview and a deep dive into PyTorch, please? Thank you!

**Comment by tienanh109** [comment]
To master PyTorch, especially given your background in machine learning (ML) and deep learning (DL) algorithms, here are some excellent resources that provide both a broad overview and a deep dive into the framework:

### Books
1. **Deep Learning with PyTorch** by Eli Stevens, Luca Antiga, and Thomas Viehmann
   - A comprehensive guide that covers the basics and more advanced topics with hands-on examples.

2. **Programming PyTorch for Deep Learning** by Ian Pointer
   - A practical book that focuses on getting started with PyTorch and implementing real-world projects.

3. **PyTorch 1.x Reinforcement Learning Cookbook** by Yuxi (Hayden) Liu
   - Focuses on reinforcement learning with practical examples and recipes.

### Online Courses
1. **Deep Learning with PyTorch: A 60 Minute Blitz** - Official PyTorch Tutorial
   - A quick introduction to PyTorch that covers the basics in a short amount of time.
   - [Link](https://pytorch.org/tutorials/beginner/deep_learning_60min_blitz.html)

2. **Deep Learning Specialization by Andrew Ng on Coursera**
   - While not solely focused on PyTorch, it's an excellent series of courses on deep learning. Pair it with PyTorch implementations of the exercises.
   - [Link](https://www.coursera.org/specializations/deep-learning)

3. **PyTorch for Deep Learning and Computer Vision on Udemy** by Jose Portilla
   - A comprehensive course that covers deep learning and computer vision applications with PyTorch.
   - [Link](https://www.udemy.com/course/pytorch-for-deep-learning-and-computer-vision/)

### Video Tutorials
1. **PyTorch Official YouTube Channel**
   - A great resource for official tutorials and updates.
   - [Link](https://www.youtube.com/c/PyTorch)

2. **Sentdex's PyTorch Tutorial Series**
   - An in-depth series of video tutorials that cover various aspects of PyTorch.
   - [Link](https://www.youtube.com/playlist?list=PLQVvvaa0QuDcNK5GeCQnxYnSSaar2tpku)

### Documentation and Tutorials
1. **PyTorch Official Documentation**
   - The go-to resource for all PyTorch-related information, including tutorials and API references.
   - [Link](https://pytorch.org/docs/stable/index.html)

2. **PyTorch Tutorials**
   - A collection of tutorials ranging from beginner to advanced levels.
   - [Link](https://pytorch.org/tutorials/)

### GitHub Repositories
1. **PyTorch Examples**
   - A repository containing various PyTorch examples, including basic and advanced models.
   - [Link](https://github.com/pytorch/examples)

2. **Awesome PyTorch**
   - A curated list of tutorials, projects, and other resources related to PyTorch.
   - [Link](https://github.com/bharathgs/Awesome-pytorch-list)

### Communities and Forums
1. **PyTorch Forums**
   - Engage with the PyTorch community to ask questions, share knowledge, and get updates.
   - [Link](https://discuss.pytorch.org/)

2. **Reddit - r/MachineLearning and r/PyTorch**
   - Subreddits where you can find discussions, news, and resources related to machine learning and PyTorch.
   - [r/MachineLearning](https://www.reddit.com/r/MachineLearning/)
   - [r/PyTorch](https://www.reddit.com/r/PyTorch/)

These resources should provide a solid foundation and help you deepen your understanding of PyTorch.

**Comment by thiunuwan** [comment]
Hi there !  Here are some resources

- Online Courses:
  - [Deep Learning with PyTorch](https://www.coursera.org/learn/deep-neural-networks-with-pytorch) on Coursera by the University of Michigan.A well-structured course from basics to more advanced topics.
   - [Introduction to Deep Learning with PyTorch on edX by IBM](https://www.edx.org/learn/deep-learning/ibm-deep-learning-with-python-and-pytorch)Offers a strong foundation in PyTorch and deep learning concepts.

- Tutorials and Documentation:
   - [Official PyTorch Documentation](https://pytorch.org/docs/stable/index.html)

**Reply by catunderrain** [reply]
Thank you so much for you suggestion!
